### ISSUE TITLE
Allow encrypted backup without password protecting the app

### ISSUE THREAD

**Issue by tbridges42** [beemdevelopment/Aegis#1345]
The use case: In my risk profile I am willing to accept the risk that my device is stolen and unlocked, however I am not willing to accept the risk that my tokens become inaccessible, nor the risk that my off-device backups are exposed in a hack. Additionally, I have to use TOTPs frequently, sometimes in time-critical situations, which makes entering a code or biometrics frustrating.

The solution: Allow the user to specify a password or biometrics for vault backups separately from specifying a password for app access. This is not less secure than the app's default configuration of no encryption at all.

I've had a look through your code and although I haven't done Android development in a long time I think I could take a stab at it if this proposal is greenlit.

**Comment by alexbakker** [comment]
This is already an option when creating a manual export, but it is not an option for automatic backups. The reason for this is the fact that we never store the user's password or the key derived from it.

We would have to make an exception in order to support automatic password protected backups for plain text Aegis vaults. I'm not sure this use case is common enough to justify it.

**Comment by ossilator** [comment]
going by the logic of the OP, the android key store should be secure enough for that.

**Comment by JoaoAparicio** [comment]
@alexbakker Would you object to this being done by public key encryption? (This is what andOTP did.)

You said 

> I'm not sure this use case is common enough to justify it.

I expect that the case "I want to be able to open the app many times a day but I want my automatic backups to have very strong encryption" should be very common, no?

**Comment by alexbakker** [comment]
Yes, because that would make the feature even more niche. We've also rejected requests for adding OpenPGP support in the past.

> I expect that the case "I want to be able to open the app many times a day but I want my automatic backups to have very strong encryption" should be very common, no?

Sure. This is why most users use the biometric unlock option. Another option is to set a very simple password (or PIN, even) to unlock the app, and a much stronger password for backups (i.e. the "Separate password for backups & exports" setting)

**Comment by Ajedi32** [comment]
FWIW this seems similar to #212 in spirit, particularly regarding "I want to be able to open the app many times a day but I want my automatic backups to have very strong encryption"
